src/volume-profile.ts

```diff
@@ -0,0 +1,129 @@
+import { Candle } from '@debut/types';
+
+type VolumeProfileData = {
+    [key: string]: VolumeItem;
+};
+
+type VolumeItem = {
+    bearish: number;
+    bullish: number;
+    total: number;
+    price: number;
+};
+
+const DEFAULT_VOLUME_ITEM: VolumeItem = {
+    bearish: 0,
+    bullish: 0,
+    price: 0,
+    total: 0,
+};
+
+export enum VolumeTypes {
+    TOTAL = 'total',
+    BULLISH = 'bullish',
+    BEARISH = 'bearish',
+}
+
+export enum PriceSource {
+    OPEN = 0,
+    CLOSE = 1,
+    LOW = 2,
+    HIGH = 3,
+    HL = 4,
+    OHLC = 5,
+}
+
+const getPriceBySource = (source: PriceSource, candle: Candle) => {
```

Comment:
I'll refactor it, this should be outside of indicator, developer must provide pre-configured source what he(she) needed.